src/system/AdminModule/Controller/AdminInterfaceController.php

```diff
@@ -0,0 +1,773 @@
+     *
+     * @return Response symfony response object
+     */
+    public function moduleheaderAction()
+    {
+    	$master_request = $this->get('request_stack')->getMasterRequest();
+    	$caller = array();
+    	$caller['_zkModule'] = $master_request->attributes->get('_zkModule');
+    	$caller['_zkType'] = $master_request->attributes->get('_zkType');
+    	$caller['_zkFunc'] = $master_request->attributes->get('_zkFunc');
+    	$caller['info']= ModUtil::getInfoFromName($caller['_zkModule']); 
+    	
+    	return ['caller' => $caller, 	
+    			'setpagetitle' => false,
+    			'insertstatusmsg' => false,
+    			'userthemename' => false,
+    			'menufirst' => false,
+    			'type' => $master_request->attributes->get('_zkType'),
+    			'image' => ModUtil::getModuleImagePath($caller['_zkModule']),
+    			'title' => $caller['info']['displayname']
+    	];
+    }
+    
+    /**
+     * @Route("/modulelinks")
+     *
+     * @Template()
+     * @Theme()
+     *
+     * Open the admin container
+     *
+     * @return Response symfony response object
+     */
+    public function modulelinksAction()
+    {
+    	$master_request = $this->get('request_stack')->getMasterRequest();
+
+    	$params['modname'] = $master_request->attributes->get('_zkModule');
+    	$caller['_zkType'] = $master_request->attributes->get('_zkType');
+    	$caller['_zkFunc'] = $master_request->attributes->get('_zkFunc');
+    	    	
+    	
+    	$menuLinks          = isset($params['links'])       ? $params['links'] : '';
+    	$menuId             = isset($params['menuid'])      ? $params['menuid'] : '';
+    	$menuClass          = isset($params['menuclass'])   ? $params['menuclass'] : 'navbar navbar-default navbar-modulelinks navbar-modulelinks-main';
+    	$menuItemClass      = isset($params['itemclass'])   ? $params['itemclass'] : '';
+    	$menuItemFirst      = isset($params['first'])       ? $params['first'] : '';
+    	$menuItemLast       = isset($params['last'])        ? $params['last'] : '';
+    	$returnAsArray      = isset($params['returnAsArray']) ? (boolean) $params['returnAsArray'] : false;
+    	
+    	if (empty($menuLinks)) {
+    		if (!isset($params['modname']) || !ModUtil::available($params['modname'])) {
+    			$params['modname'] = ModUtil::getName();
+    		}
+    	
+    		// check our module name
+    		if (!ModUtil::available($params['modname'])) {
+    		//	$view->trigger_error('modulelinks: '.__f("Error! The '%s' module is not available.", DataUtil::formatForDisplay($params['modname'])));
+    	
+    			return false;
+    		}
+    	
+    		$params['type'] = isset($params['type']) ? $params['type'] : 'admin';
+    	
+    		// get the menu links
+    		// try the Core-2.0 way first, then try the legacy way.
+    		$menuLinks = $this->get('zikula.link_container_collector')->getLinks($params['modname'], $params['type']);
+    		if (empty($menuLinks)) {
+    			$menuLinks = ModUtil::apiFunc($params['modname'], $params['type'], 'getLinks', $params);
```

Comment:
mark code that must be removed with something like:
`// @todo - remove this BC code at Core-2.0`

the `Core-2.0` marker is important because I will be doing a global search for all such markers when the master branch is revived with no BC code.